src/libMessage/Messenger.cpp

```diff
@@ -4808,16 +4808,14 @@ bool Messenger::SetNodeForwardTxnBlock(
   result.set_shardid(shardId);
   SerializableToProtobufByteArray(lookupKey.second, *result.mutable_pubkey());
 
-  unsigned int txnsCurrentCount = 0;
-  unsigned int txnsGeneratedCount = 0;
-
-  unsigned int msg_size = 0;
+  unsigned int txnsCurrentCount = 0, txnsGeneratedCount = 0, msg_size = 0;
 
+  const auto protoTxn = std::make_unique<ProtoTransaction>();
   for (const auto& txn : txnsCurrent) {
     if (msg_size >= PACKET_BYTESIZE_LIMIT) {
       break;
     }
-    ProtoTransaction* protoTxn = new ProtoTransaction();
+
     TransactionToProtobuf(txn, *protoTxn);
```

Comment:
https://github.com/Zilliqa/Zilliqa/blob/b65aeecb8c14e2d7198bbfd3ad295501930d05de/src/libMessage/Messenger.cpp#L1143

https://github.com/Zilliqa/Zilliqa/blob/b65aeecb8c14e2d7198bbfd3ad295501930d05de/src/libMessage/Messenger.cpp#L1157

If we reuse the same pointer,  the data and/or code from previous txn can be set for next txn??
or else need some way to clear ProtoTransaction before reusing it.

Reply:
hmm. If it is the case, we can reset the pointer to nullptr first?

Reply:
Probably better to keep the smart pointer initialization in the same line within the for loop.

Reply:
yes better to initialize in loop itself.

Reply:
TransactionToProtobuf should always reinitialize the whole buffer, so the pointer should can be reused. Can add a test to verify it.

Reply:
Hmm nope not completely true, because there are optional fields like `code` and `data` which may have been set in the previous loop iteration but unmodified in the current iteration.

Reply:
Good point

Reply:
@ansnunez Updated. Please review again.